cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,185 @@
+  const CatalogRelation *relation;
+  if (arguments->size() == 0) {
+    for (const CatalogRelation &rel : catalog_database) {
+      int column_width = static_cast<int>(rel.getName().length());
+      max_column_width =
+          max_column_width < column_width ? column_width : max_column_width;
+    }
+  } else {
+    const ParseString &table_name = arguments->front();
+    const std::string &table_name_val = table_name.value();
+    relation = catalog_database.getRelationByName(table_name_val);
+
+    if (relation == nullptr) {
+     THROW_SQL_ERROR_AT(&(arguments->front())) << " Unrecognized relation "  <<table_name_val;
+    }
+    int column_width = static_cast<int>(relation->getName().length());
+    max_column_width =
+        max_column_width < column_width ? column_width : max_column_width;
+  }
+  // Only if we have relations work on the printing logic.
+  if (catalog_database.size() > 0) {
+    vector<int> column_widths;
+    column_widths.push_back(max_column_width+1);
+    column_widths.push_back(C::kInitMaxColumnWidth+1);
+    fputs("       List of relations\n\n", out);
+    fprintf(out, "%-*s |", max_column_width+1, " Name");
+    fprintf(out, "%-*s\n", C::kInitMaxColumnWidth, " Type");
+    PrintToScreen::printHBar(column_widths, out);
+    //  If there are no argument print the entire list of tables
+    //  else print the particular table only.
+    if (arguments->size() == 0) {
+      for (const CatalogRelation &rel : catalog_database) {
+        fprintf(out, " %-*s |", max_column_width, rel.getName().c_str());
+        fprintf(out, " %-*s\n", C::kInitMaxColumnWidth, "table");
+      }
+    } else {
+      fprintf(out, " %-*s |", max_column_width, relation->getName().c_str());
+      fprintf(out, " %-*s\n", C::kInitMaxColumnWidth, "table");
+    }
+    fputc('\n', out);
+  }
+}
+
+void executeDescribeTable(
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase &catalog_database, FILE *out) {
+  const ParseString &table_name = arguments->front();
+  const std::string &table_name_val = table_name.value();
+  const CatalogRelation *relation =
+      catalog_database.getRelationByName(table_name_val);
+  if (relation == nullptr) {
+     THROW_SQL_ERROR_AT(&(arguments->front())) << " Unrecognized relation "  <<table_name_val;
+  }
+  vector<int> column_widths;
+  int max_attr_column_width = C::kInitMaxColumnWidth;
+  int max_type_column_width = C::kInitMaxColumnWidth;
+
+  for (const CatalogAttribute &attr : *relation) {
+    // Printed column needs to be wide enough to print:
+    //   1. The attribute name (in the printed "header").
+    //   2. Any value of the attribute's Type.
+    int attr_column_width = attr.getDisplayName().length();
+    int type_column_width = attr.getType().getName().length();
+    max_attr_column_width = std::max(max_attr_column_width, attr_column_width);
+    max_type_column_width = std::max(max_type_column_width, type_column_width);
```

Comment:
Please replace `attr_column_width` and `type_column_width` with their values in the assignment expressions for `max_attr_column_width` and `max_type_column_width`, respectively.

Otherwise, mark the former two `const`.